### ISSUE TITLE
As an Open Data Hub Editor I want to edit also GPS coordinates, to improve the position of my POIs

### ISSUE THREAD

**Issue by sseppi** [noi-techpark/it.bz.opendatahub.databrowser#360]
As a first step, I would suggest to implement the possibility to change the GPS coordinates by imputing the Logitude and Latitude and other information in the proper input fields.

The possibility to change the position by using the map will be postponed to the next iteration and, in case, a dedicated issue will be created.

**Comment by sseppi** [comment]
@pkritzinger @Mazzolintis we need to discuss this topic during the next meeting, we need to find a way to edit GPS coordinates and then implement it.

**Comment by sseppi** [comment]
@pkritzinger can you please share the link to the design, you showed during the meeting?

The Altitude field will be set as Read Only in the API and shown as a non editable field in the Data Browser. We need to remove the drop-down icon.

For the GPS-Type the component drop-down+freetext will be used. The first GPS inserted per default will be GPS-Type = position. This aspect will be further clarified with Rudi.

**Comment by pkritzinger** [comment]
@sseppi find the link here:

https://www.figma.com/file/DmuP6Dbv5LzkCNOXrDnWIL/2023?type=design&node-id=1502%3A9766&mode=design&t=Ha38KlFfOl45aW1j-1

**Comment by sseppi** [comment]
@pkritzinger thank you!

**Comment by sseppi** [comment]
@MatteoBiasi @gappc 

Me and Martin had a talk with Rudi about the ```Gpstype``` field and the outcome was the following.

We decided to keep using only the ```GpsInfo``` attribute and deprecate the ```GpsPoints``` since the first one is more easy to use and understand for the API consumers.

Considering this decision, to render and populate the GPS coordinates of an element, we should use only the ```GpsInfo``` attribute and the Open Data Hub will then sync the data also in the ```GpsPoints``` deprecated attribute. This means that the ```Gpstype``` field becomes optional and in case that the user does not populate it the Open Data Hub will assign a default ```Gpstype``` while he does the Sync between ```GpsInfo``` and ```GpsPoints``` attribute.

**Comment by gappc** [comment]
@sseppi @RudiThoeni to clearify, you mean the `GpsInfo` array, that contains GPS objects? Is `Gpstype` then optional?

@sseppi could you please update the other issues, so we have one single decision that is valid for all issues?

**Comment by RudiThoeni** [comment]
is it possible to have a short call monday/Tuesday next week where we clarify this definitely?

**Comment by sseppi** [comment]
@gappc would it be possible for you to have a short call monday 30 between 9 and 11 am?

**Comment by gappc** [comment]
@RudiThoeni @sseppi monday I have time until 10:00, it would be great if we could meet before. Just let me know.

**Comment by RudiThoeni** [comment]
As mentioned in issue #435 https://github.com/noi-techpark/it.bz.opendatahub.databrowser/issues/435 we will use `GpsInfo` as the field where GPS Info is stored

**Comment by pkritzinger** [comment]
@sseppi @gappc @MatteoBiasi 
we had a look at how big players allow users to individually set a position on the map. Honestly speaking, neither on Google Maps nor on Apple Maps a clear pattern could be found. Sometimes a long-pressed click, sometimes a simple click led to the desired result. 

With Google myMaps, we found an interaction pattern that could be a best practice for our needs. We documented it in the following figma file and added our comments. What do you think? If fine for you we can create the design.
https://www.figma.com/file/DmuP6Dbv5LzkCNOXrDnWIL/2023?type=design&node-id=1582%3A293&mode=design&t=X8T3f6H4jPBwdGWW-1

**Comment by MatteoBiasi** [comment]
@pkritzinger thanks for your proposal. By my side I agree that the proposed solution is pretty consistent in terms of UX. On the other site it requires a pretty heavy effort in terms of development, particularly for the pin drag & drop feature. Unless we can find something already done, I believe we need to assess if it's worth it, or if a simple click on the map to set the pin and move it by clicking elsewhere is sufficient. Drag & Drop would become useful if we had more pins, but at the moment, that's not our case.

**Comment by sseppi** [comment]
Since in the future we plan o remove the Edit View and provide the possibility to activate the editing for the single fields, we decided to test this behaviour on the map.

Next Steps
- @pkritzinger will provide the design of a button to activate the editing of the maps that could be used also for the other fields;
- @MatteoBiasi will implement the solution

The map, also in edit mode, will be kapt small and ther will be an icon on the map to open the full screen mode.

**Comment by pkritzinger** [comment]
@sseppi @MatteoBiasi please find the final design here: 
https://www.figma.com/file/DmuP6Dbv5LzkCNOXrDnWIL/2023?type=design&node-id=1582%3A293&mode=design&t=a5FLGbJKDv9XUu40-1

Comments describe the behavior of the UI.

**Comment by sseppi** [comment]
@pkritzinger thank you for the design!

@MatteoBiasi can you please implement also the changes (add the two buttons) on the full screen mode for the images?

**Comment by RudiThoeni** [comment]
I merged the latest PR @sseppi please test, i let you close the issue.

Very good component, good work 👍  I tested and was able to add/modify gps points on an accommodation and an activity

**Comment by sseppi** [comment]
I tested the new functionality too, and it seems to work fine. I have only one questo for @pkritzinger.

If I'm in the `Edit View` in the GPS section and I open the full screen map from the table (see screenshot below)

![Schermata da 2023-12-15 17-29-37](https://github.com/noi-techpark/it.bz.opendatahub.databrowser/assets/7909989/87065ea8-c83b-4b77-8b78-da9444ee3d97)

Wouldn't be better, to open the editable map (screenshot below)? In the actual version online in testing the map without the possibility to change the GPS point get open.

![Schermata da 2023-12-15 17-33-02](https://github.com/noi-techpark/it.bz.opendatahub.databrowser/assets/7909989/3e2117b6-17ce-4d9a-b9e1-8e7f459e2879)

What is your feedback?

**Comment by pkritzinger** [comment]
@sseppi from my point of view it is fine as it has been implemented, since:
- behavior is the same as for the image-component
- if we changed it to your proposal the behavior of the fullscreen view would be different between edit view and detail view

what i found: 
in the detail view the user has the possibility to enter the detail of the table where the pin can be moved. This feature should be hidden. Also, by clicking on fullscreen, the map should only be opened in fullscreen without having the possibility to change the position of the pin.

![image](https://github.com/noi-techpark/it.bz.opendatahub.databrowser/assets/92454516/8e28214e-06e0-4b3b-963d-f01e116db2d9)
